modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/utils/SW360ComponentAdapterUtils.java

```diff
@@ -12,17 +12,26 @@
 
 import org.eclipse.sw360.antenna.model.artifact.Artifact;
 import org.eclipse.sw360.antenna.model.artifact.ArtifactCoordinates;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactFilename;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactFilename.ArtifactFilenameEntry;
 import org.eclipse.sw360.antenna.model.coordinates.Coordinate;
 import org.eclipse.sw360.antenna.sw360.rest.resource.components.SW360Component;
 import org.eclipse.sw360.antenna.sw360.rest.resource.components.SW360ComponentType;
 import org.eclipse.sw360.antenna.sw360.rest.resource.releases.SW360Release;
 
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
 public class SW360ComponentAdapterUtils {
     public static String createComponentName(Artifact artifact) {
         return artifact.askFor(ArtifactCoordinates.class)
                 .map(ArtifactCoordinates::getMainCoordinate)
-                .map(Coordinate::getName)
-                .orElse(artifact.toString()); // TODO: ugly hack
+                .map(o -> Stream.of(o.getNamespace(), o.getName()))
+                .map(o -> o.collect(Collectors.joining("/")))
+                .orElse(artifact.askFor(ArtifactFilename.class)
+                        .flatMap(ArtifactFilename::getBestFilenameEntryGuess)
+                        .map(ArtifactFilenameEntry::getFilename)
+                        .orElse(""));
```

Comment:
I am not really happy with an empty string. This is the name that will stand in SW360 and there shouldn't be multiple ones with empty string. Additionally by this name the release with be searched, making wrong assignments to releases with the same version and also an empty string name very likely.

Reply:
I think, if there is no name here, we maybe shouldn't have this in SW360 at all. 
@blaumeiser-at-bosch what do you think?

Reply:
That was my though as well, if we do not have enough meat to at least have a simple name, we should inform the user that there is something strange here, and we should not push arbitrary data to SW360. At least a filename should be there.

Reply:
Yes the user should be informed, but not in the adapter utils class? The component client will fail in creating a component with an empty string and will log an error.

Reply:
Do you mean when the request is made?
I feel like we should try to avoid making requests we know will fail to save resources.

Reply:
We are trying to avoid it with the `CoordinatesValidator`

Reply:
I totally agree that here is not the point of informing the user here. This is a helper who does something useful. The question is, whether no information at all should return an error aka an exception instead of an empty string?

Reply:
Just for clarity, is that the flow you want?
```
if AritfactCoordinate then
   return <component name extracted from Coordinates>
else if ArtifactFilename then
   return <component name extracted from ArtifactFilenameEntry>
else 
   throw new ExecutionException(..);
```

Reply:
Yes, I think this makes sense. This exception can be caught and it is easier to distinguish between make the right thing and do something exceptional